kubernetes-client/src/main/java/io/fabric8/kubernetes/client/BaseKubernetesClient.java

```diff
@@ -132,6 +127,21 @@
  * It is thread safe.
  */
 public abstract class BaseKubernetesClient<C extends Client> extends BaseClient implements GenericKubernetesClient<C> {
+  
+  static {
+    Handlers.register(Pod.class, PodOperationsImpl::new);
```

Comment:
Perhaps the `Handlers` class could be generated during the build process so that we don't need to manually register new ones?

Reply:
There are a couple of approaches to be weighed:

- something generated - but that will need the some templating mechanism, which we should otherwise be able to remove
- just a reflective association by convention similar to List and Builder classes - bit it seemed more involved to determine the package name, so I didn't take that approach yet.  There is some validation logic in my next commit thought that could easily determine if an OperationsImpl is not being picked up by the Handler system that would be applicable here.
- more logic around the initialization - this is what I've started to added in the next commit.  If the *ExtensionAdapter class uses references the *Client then that can trigger the static block to register the handlers.  Alternatively that registration logic could be moved onto the *ExtensionAdapter (which is probably a better option than the former).